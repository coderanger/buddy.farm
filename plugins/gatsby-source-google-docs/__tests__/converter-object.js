const fs = require("fs")
const path = require(`path`)

const {GoogleDocument} = require("../utils/google-document")

const documentsPath = path.join(__dirname, "documents")
const filenames = fs.readdirSync(documentsPath)

filenames.forEach(function (filename) {
  const filepath = path.join(documentsPath, filename)
  const file = fs.readFileSync(filepath, "utf8")
  const document = JSON.parse(file)
  const googleDocument = new GoogleDocument({document})

  test(`Document "${googleDocument.document.title}" to Object`, () => {
    const {cover, elements} = googleDocument

    expect({
      cover,
      elements,
    }).toMatchSnapshot()
  })
})
