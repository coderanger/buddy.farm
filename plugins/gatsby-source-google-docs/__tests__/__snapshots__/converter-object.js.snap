// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document "Breaks" to Object 1`] = `
Object {
  "cover": null,
  "elements": Array [],
}
`;

exports[`Document "Codes" to Object 1`] = `
Object {
  "cover": null,
  "elements": Array [
    Object {
      "type": "p",
      "value": "Type \`gatsby-source-google-docs-token\` to generate a token",
    },
    Object {
      "type": "code",
      "value": Object {
        "content": Array [
          "",
          "const foo = \\"bar\\"",
          "",
          "module.exports = {",
          "    plugins: [",
          "        \\"gatsby-source-google-docs\\",",
          "    ],",
          "}",
        ],
        "language": "js",
      },
    },
    Object {
      "type": "code",
      "value": Object {
        "content": Array [
          "module.exports = {",
          "    plugins: [",
          "        \\"gatsby-source-google-docs\\",",
          "    ],",
          "}",
        ],
        "language": null,
      },
    },
  ],
}
`;

exports[`Document "Cover" to Object 1`] = `
Object {
  "cover": Object {
    "alt": "Cover description",
    "image": "https://lh3.googleusercontent.com/nDkE8IlmeehGxZUwGXvxn_dMEnZGZBFAp0i3p_07UC9Lj_V_wfNCBh3FbcHiknyR_aNXue63CmGPi9dTykd9gqobAu88AgHq1af0ag81US_ykXwb-gm04enjY_feYOfPTD3Biq1fk03IKvZ8RA",
    "title": "Cover title",
  },
  "elements": Array [],
}
`;

exports[`Document "Empty" to Object 1`] = `
Object {
  "cover": null,
  "elements": Array [],
}
`;

exports[`Document "Footnotes" to Object 1`] = `
Object {
  "cover": null,
  "elements": Array [
    Object {
      "type": "p",
      "value": "Text with a footnote[^1] and an other one[^2]",
    },
    Object {
      "type": "footnote",
      "value": Object {
        "number": "1",
        "text": " Footnote 1 description",
      },
    },
    Object {
      "type": "footnote",
      "value": Object {
        "number": "2",
        "text": " Footnote 2 description",
      },
    },
  ],
}
`;

exports[`Document "French" to Object 1`] = `
Object {
  "cover": null,
  "elements": Array [
    Object {
      "type": "p",
      "value": "Je m'appelle **CÃ©dric**, j'habite Ã  Toulouse _(France)_ ðŸ‡«ðŸ‡·",
    },
  ],
}
`;

exports[`Document "Horizontal rule" to Object 1`] = `
Object {
  "cover": null,
  "elements": Array [
    Object {
      "type": "p",
      "value": "Text followed by horizontal rule ",
    },
    Object {
      "type": "p",
      "value": "<hr/>",
    },
    Object {
      "type": "p",
      "value": "and some other text",
    },
  ],
}
`;

exports[`Document "Images" to Object 1`] = `
Object {
  "cover": null,
  "elements": Array [
    Object {
      "type": "img",
      "value": Object {
        "alt": "",
        "source": "https://lh6.googleusercontent.com/pZBSuNhNbnEaBQZSfpSZPp8k2Obc7TSMFrhUx5pvcMP9v3EkBnn7EQsnKftFH6HaFWKDZR_zUlusaPQ6DULiSTIsUHd1PW5hdmuNaF1cVifWwB3reUL2spqfO7qyWGjT6x4J1MkwtmDKayvKgw",
        "title": "",
      },
    },
  ],
}
`;

exports[`Document "Links" to Object 1`] = `
Object {
  "cover": null,
  "elements": Array [
    Object {
      "type": "p",
      "value": "[to self](https://docs.google.com/document/d/1UuBxtIEEVh98wyBR9fMmLqzJEkmNlAMMoC4SEhprHfQ)",
    },
    Object {
      "type": "p",
      "value": "[to self with user id](https://docs.google.com/document/u/1/d/1UuBxtIEEVh98wyBR9fMmLqzJEkmNlAMMoC4SEhprHfQ)",
    },
    Object {
      "type": "p",
      "value": "[to self with edit](https://docs.google.com/document/d/1UuBxtIEEVh98wyBR9fMmLqzJEkmNlAMMoC4SEhprHfQ/edit)",
    },
    Object {
      "type": "p",
      "value": "[to self with preview](https://docs.google.com/document/d/1UuBxtIEEVh98wyBR9fMmLqzJEkmNlAMMoC4SEhprHfQ/preview)",
    },
    Object {
      "type": "p",
      "value": "[unknown url](https://docs.google.com/document/d/unknown)",
    },
  ],
}
`;

exports[`Document "Lists" to Object 1`] = `
Object {
  "cover": null,
  "elements": Array [
    Object {
      "type": "ul",
      "value": Array [
        "List **item** 1",
        Object {
          "type": "ol",
          "value": Array [
            "Sublist item 1",
            "Sublist item 2",
            "Sublist item 3",
          ],
        },
        "<ins>List</ins> item 2",
        "List item 3",
        Object {
          "type": "ol",
          "value": Array [
            "Sub list item 1",
            Object {
              "type": "ol",
              "value": Array [
                "Sub sub list item 1",
                "Sub sub list item 2 with ![](https://lh4.googleusercontent.com/IqgS1qr6oFq7jaAsQbCkfJsuwBg5DnKfnjTuB49VbHQr0ZN8T1UjgRDoiQzPws-hWgJcRDyujobZmOEF7BAHwceM6kTrXC_2dj-E38kNxmz8EW59q1ZvsM0C9cUxytASTReIN13NA8fcFA8zIA \\"\\")image",
              ],
            },
            "Sub list item 2",
            "Sub list item 3",
          ],
        },
        "List item 4",
        "List item 5",
      ],
    },
  ],
}
`;

exports[`Document "Poem" to Object 1`] = `
Object {
  "cover": null,
  "elements": Array [
    Object {
      "type": "p",
      "value": "The poem",
    },
    Object {
      "type": "p",
      "value": "    may            have",
    },
    Object {
      "type": "p",
      "value": "              strange whitespace",
    },
    Object {
      "type": "p",
      "value": "patterns.",
    },
  ],
}
`;

exports[`Document "Quotes" to Object 1`] = `
Object {
  "cover": null,
  "elements": Array [
    Object {
      "type": "blockquote",
      "value": "\\"The way to get started is to quit talking and begin doing.\\"",
    },
  ],
}
`;

exports[`Document "Special characters" to Object 1`] = `
Object {
  "cover": null,
  "elements": Array [
    Object {
      "type": "p",
      "value": "<span style='color:rgb(51, 51, 51);background-color:rgb(255, 255, 255)'>19980006564â€¯-â€¯0081</span>",
    },
  ],
}
`;

exports[`Document "Tables" to Object 1`] = `
Object {
  "cover": null,
  "elements": Array [
    Object {
      "type": "table",
      "value": Object {
        "headers": Array [
          "Col 1",
          "Col 2",
          "Col 3",
        ],
        "rows": Array [
          Array [
            "Col 1 line 1",
            "Col 2 line 1",
            "Col 3 line 1",
          ],
          Array [
            "Col 1 line 2",
            "Col 2 line 2",
            "Col 3 line 2",
          ],
        ],
      },
    },
  ],
}
`;

exports[`Document "Texts" to Object 1`] = `
Object {
  "cover": null,
  "elements": Array [
    Object {
      "type": "h2",
      "value": "Title level 1",
    },
    Object {
      "type": "h3",
      "value": "Title level 2",
    },
    Object {
      "type": "h4",
      "value": "Title level 3",
    },
    Object {
      "type": "h5",
      "value": "Title level 4",
    },
    Object {
      "type": "h6",
      "value": "Title level 5",
    },
    Object {
      "type": "h6",
      "value": "Title level 6",
    },
    Object {
      "type": "h2",
      "value": "Title",
    },
    Object {
      "type": "h3",
      "value": "Subtitle",
    },
    Object {
      "type": "p",
      "value": "**bold**",
    },
    Object {
      "type": "p",
      "value": "_italic_",
    },
    Object {
      "type": "p",
      "value": "<ins>underline</ins>",
    },
    Object {
      "type": "p",
      "value": "~~strikethrough~~",
    },
    Object {
      "type": "p",
      "value": "<sup>superscript</sup>",
    },
    Object {
      "type": "p",
      "value": "<sub>subscript</sub>",
    },
    Object {
      "type": "p",
      "value": "**_boldItalic_**",
    },
    Object {
      "type": "p",
      "value": "text with **space bold after** and _space italic before_.",
    },
    Object {
      "type": "p",
      "value": "		indented text",
    },
    Object {
      "type": "p",
      "value": "vertical<br/>indent",
    },
    Object {
      "type": "p",
      "value": "[Link](https://github.com/cedricdelpoux/gatsby-source-google-docs)",
    },
    Object {
      "type": "p",
      "value": "<span style='font-size:1.09em'>Text bigger</span>",
    },
    Object {
      "type": "p",
      "value": "<span style='color:rgb(106, 168, 79)'>Text Colored</span><span style='color:rgb(255, 0, 0)'>red</span><span style='color:rgb(0, 255, 0)'>green</span><span style='color:rgb(0, 0, 255)'>blue</span>",
    },
    Object {
      "type": "p",
      "value": "<span style='background-color:rgb(147, 196, 125)'>Text with background color</span><span style='background-color:rgb(255, 0, 0)'>red</span><span style='background-color:rgb(0, 255, 0)'>green</span><span style='background-color:rgb(0, 0, 255)'>blue</span>",
    },
  ],
}
`;
